libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -506,3 +534,97 @@ def test_that_case_sensitivity_does_not_affect_distance_strategy(
     conn.execute(text("use master"))
     conn.execute(text(_DROP_COLLATION_DB_QUERY))
     conn.close()
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_1_FILTERING_TEST_CASES)
+def test_sqlserver_with_with_metadata_filters_1(
+    store: SQLServer_VectorStore,
+    test_filter: Dict[str, Any],
+    expected_ids: List[int],
+) -> None:
+    store.add_texts(filter_texts, filter_metadatas, filter_ids)
+
+    docs = store.similarity_search("meow", k=5, filter=test_filter)
+    store.delete(["1", "2", "3"])
+    returned_ids = [doc.metadata["id"] for doc in docs]
+    assert sorted(returned_ids) == sorted(expected_ids), test_filter
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_2_FILTERING_TEST_CASES)
+def test_sqlserver_with_with_metadata_filters_2(
+    store: SQLServer_VectorStore,
+    test_filter: Dict[str, Any],
+    expected_ids: List[int],
+) -> None:
+    store.add_texts(filter_texts, filter_metadatas, filter_ids)
+
+    docs = store.similarity_search("meow", k=5, filter=test_filter)
+    store.delete(["1", "2", "3"])
+    returned_ids = [doc.metadata["id"] for doc in docs]
+    assert sorted(returned_ids) == sorted(expected_ids), test_filter
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_3_FILTERING_TEST_CASES)
+def test_sqlserver_with_with_metadata_filters_3(
+    store: SQLServer_VectorStore,
+    test_filter: Dict[str, Any],
+    expected_ids: List[int],
+) -> None:
+    store.add_texts(filter_texts, filter_metadatas, filter_ids)
+
+    docs = store.similarity_search("meow", k=5, filter=test_filter)
+    store.delete(["1", "2", "3"])
+    returned_ids = [doc.metadata["id"] for doc in docs]
+    assert sorted(returned_ids) == sorted(expected_ids), test_filter
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_4_FILTERING_TEST_CASES)
+def test_sqlserver_with_with_metadata_filters_4(
+    store: SQLServer_VectorStore,
+    test_filter: Dict[str, Any],
+    expected_ids: List[int],
+) -> None:
+    store.add_texts(filter_texts, filter_metadatas, filter_ids)
+
+    docs = store.similarity_search("meow", k=5, filter=test_filter)
+    store.delete(["1", "2", "3"])
+
+    returned_ids = [doc.metadata["id"] for doc in docs]
+    assert sorted(returned_ids) == sorted(expected_ids), test_filter
+
+
+@pytest.mark.parametrize("test_filter, expected_ids", TYPE_5_FILTERING_TEST_CASES)
```

Comment:
except for the test_filter parameter, all the tests have the exact same code, cant we write a single test or a single function that does the test and take input the test filter?